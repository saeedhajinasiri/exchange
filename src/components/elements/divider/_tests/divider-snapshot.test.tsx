import React from "react";
import {cleanup, render} from "test";
import Divider from "../divider";

afterEach(cleanup)

describe('Test Divider UI', () => {
  it('Default', () => {
    const {asFragment} = render(<Divider/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Divider Colors', () => {
  it('color=primary', () => {
    const {asFragment} = render(<Divider color={"primary"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=secondary', () => {
    const {asFragment} = render(<Divider color={"secondary"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=tertiary', () => {
    const {asFragment} = render(<Divider color={"tertiary"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=grey', () => {
    const {asFragment} = render(<Divider color={"grey"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=light', () => {
    const {asFragment} = render(<Divider color={"light"}/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=dark', () => {
    const {asFragment} = render(<Divider color={"dark"}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

