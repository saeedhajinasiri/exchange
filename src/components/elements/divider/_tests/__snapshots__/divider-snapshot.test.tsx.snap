// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Divider Colors color=dark 1`] = `
<DocumentFragment>
  <div
    class="w-full border-t my-10 opacity-30 border-t-grey-dark"
  />
</DocumentFragment>
`;

exports[`Test Divider Colors color=grey 1`] = `
<DocumentFragment>
  <div
    class="w-full border-t my-10 opacity-30 border-t-grey"
  />
</DocumentFragment>
`;

exports[`Test Divider Colors color=light 1`] = `
<DocumentFragment>
  <div
    class="w-full border-t my-10 opacity-30 border-t-grey-light"
  />
</DocumentFragment>
`;

exports[`Test Divider Colors color=primary 1`] = `
<DocumentFragment>
  <div
    class="w-full border-t my-10 opacity-30 border-t-primary\`"
  />
</DocumentFragment>
`;

exports[`Test Divider Colors color=secondary 1`] = `
<DocumentFragment>
  <div
    class="w-full border-t my-10 opacity-30 border-t-secondary"
  />
</DocumentFragment>
`;

exports[`Test Divider Colors color=tertiary 1`] = `
<DocumentFragment>
  <div
    class="w-full border-t my-10 opacity-30 border-t-tertiary"
  />
</DocumentFragment>
`;

exports[`Test Divider UI Default 1`] = `
<DocumentFragment>
  <div
    class="w-full border-t my-10 opacity-30 border-t-grey-light"
  />
</DocumentFragment>
`;
